metasmoke.py

```diff
@@ -32,16 +32,103 @@
 import metasmoke_cache
 
 
-MAX_FAILURES = 10  # Preservative, 10 errors = MS down
 NO_ACTIVITY_PINGS_TO_STANDBY = 8
 NO_ACTIVITY_PINGS_TO_REPORT = 4
 
 
 # noinspection PyClassHasNoInit,PyBroadException,PyUnresolvedReferences,PyProtectedMember
 class Metasmoke:
     status_pings_since_scan_activity = 0
-    last_ping_scan_time = 0
-    last_ping_posts_scanned = 0
+
+    class AutoSwitch:
+        """ Automatically switch metasmoke status """
+        # Last change was on 20 May 2020.
+        MAX_FAILURES = 10  # 10 failures == ms down
+        MAX_SUCCESSES = 1  # 1 success == ms up
+        counter = 0
```

Comment:
How about `failure_count`?

Reply:
The issue is that the counter is used for both success and failure (and we don't want super long names like `ping_success_failure_counter`)

Reply:
(and that variable is private anyway)

Reply:
I'd argue that a verbose name is better than one which is short enough to be unclear what it does. I'm fine with `ping_failure_count`, `net_failure_count`, etc. One of the points of having descriptive variable names is so that code is more self-documenting and requires fewer comments to guide the reader's understanding of what's being done and why. I probably wouldn't use `ping_success_failure_count`, because having success in the name may make it seem that it should be positive for successes, whereas a negative number in a `failure_count` is hard to interpret as something other than successes.

Reply:
That it's private, doesn't really matter. The point of having variable names be descriptive is so that when the next programmer comes through needing to change something, even if that's you a couple years from now, they know exactly what the variable is doing, instead of seeing `counter` and then thinking "counter? what counter? what's it counting?"